server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1462,12 +1480,12 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
             }
         }
 
-        if (ntwkOff.getGuestType() != GuestType.Shared && (!StringUtils.isAllBlank(routerIp, routerIpv6))) {
-            throw new InvalidParameterValueException("Router IP can be specified only for Shared networks");
+        if (ntwkOff.getGuestType() == GuestType.L2 && (!StringUtils.isAllBlank(cmd.getRouterIPv4(), cmd.getRouterIPv6()))) {
```

Comment:
Is there a validation for isolated networks source 
nat support and capabilities here ?

Reply:
There isn´t one, but I think supported services is handled in another location. in my tests I've seen no problems so far. Also this change is only a small fix, allowing the router ip to be specified, which would already happen if you don´t start and isolated network and then allocate an new IP.

Maybe there is another check you are missing?

Reply:
@DaanHoogland 
I was just wondering if there are checks to verify the source nat is supported by the network (or offering) and the specified IP is available. I did not find the checks.
I will have a look again. If you know the position of checks, it would be great.

Reply:
I'll try to break it by creating an offering without source nat and see what I can break.